cmd/limactl/shell.go

```diff
@@ -108,7 +109,14 @@ func shellAction(cmd *cobra.Command, args []string) error {
 	}
 	logrus.Debugf("changeDirCmd=%q", changeDirCmd)
 
-	script := fmt.Sprintf("%s ; exec $SHELL --login", changeDirCmd)
+	shell, err := cmd.Flags().GetString("shell")
+	if err != nil {
+		return err
+	}
+	if shell == "" {
+		shell = "$SHELL"
+	}
+	script := fmt.Sprintf("%s ; exec %s --login", changeDirCmd, shell)
```

Comment:
Better to quote?

Reply:
I don't think this should ever be necessary, but why not...

I've also squeezed in unrelated calls to `shellescape.Quote` directory names. We used to use `%q` for them, but that is wrong if the name contains a `$` sign: `"foo$bar"` needs to be `'foo$bar'`.